File: Npc/11001923.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11001923: Fisher
/// </summary>
public class _11001923 : NpcScript {
    protected override int First() {
        return 30;
    }

    // Select 0:
    // $script:1121140707007424$
    // - I caught a prize fish!
    protected override int Select() => 0;

    protected override int Execute(int selection) {
        switch (Id, Index++) {
            case (30, 0):
                // $script:1121184207007433$
                // - Hey! Find your own spot!
                return -1;
        }
        
        return default;
    }

    protected override NpcTalkButton Button() {
        return (Id, Index) switch {
            (30, 0) => NpcTalkButton.Close,
            _ => NpcTalkButton.None,
        };
    }
}

File: Npc/11001923.py
""" 11001923: Fisher """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 30

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:1121140707007424$
        # - I caught a prize fish!
        return None # TODO

    def __30(self, pick: int) -> int:
        # $script:1121184207007433$
        # - Hey! Find your own spot!
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (30, 0):
            return Option.CLOSE
        return Option.NONE
